### PR TITLE
Snapshot takes picture of HMD window instead of preview

### PR DESCRIPTION
**Pull request by SamGondelman**
This addresses fogbugz case [1003](https://highfidelity.fogbugz.com/f/cases/1003/Snapshot-on-Mac-while-in-HMD-is-busted) and makes it so that when you take a snapshot in HMD mode, it takes a picture of the actual HMD window (with both eyes shown), instead of the desktop window.

### CODE REVIEW

**Comment by SamGondelman** [comment]
Testing:

- On Windows, in an HMD, take a snapshot (CTRL-S).  It should show both of your eyes and have no black borders.
- Repeat on Mac.  The snapshot should be of the HMD view regardless of if you have vsync enabled (preview broken) or disabled (preview working).

**Comment by hifi-gustavo** [comment]
Build console output for commit: 40d584021a47e2c0d1960f13a07ac98115263cf1 available at: https://gist.github.com/bc96e69154917767d8ee723ff1ff8c2e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8162.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8162.dmg

**Comment by thoys** [comment]
oops warning in Ubuntu build:
```
/data/jenkins/Jenkins/workspace/beta_pr-master/label/ubuntu/libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp: In lambda function:
/data/jenkins/Jenkins/workspace/beta_pr-master/label/ubuntu/libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp:653:14: warning: variable ‘windowSize’ set but not used [-Wunused-but-set-variable]
         auto windowSize = toGlm(_container->getPrimaryWidget()->size());
              ^
```

File: libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp [hunk 1]
```diff
@@ -647,12 +647,14 @@ void OpenGLDisplayPlugin::withMainThreadContext(std::function<void()> f) const {
 }
 
 QImage OpenGLDisplayPlugin::getScreenshot() const {
-    QImage result;
+    using namespace oglplus;
+    QImage screenshot(_compositeFramebuffer->size.x, _compositeFramebuffer->size.y, QImage::Format_RGBA8888);
     withMainThreadContext([&] {
-        static auto widget = _container->getPrimaryWidget();
-        result = widget->grabFrameBuffer();
+        auto windowSize = toGlm(_container->getPrimaryWidget()->size());
```
Comment (thoys) [inline libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp]: windowSize is unused, that fails the build on Ubuntu

Comment (SamGondelman) [inline libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp]: haha yeah I realized

**Comment by hifi-gustavo** [comment]
Build console output for commit: be102ba9265a8915ec947414bae05fb43933704f available at: https://gist.github.com/bc96e69154917767d8ee723ff1ff8c2e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8162.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8162.dmg

**Comment by OmegaHeron** [comment]
QA Pass
Testing Windows 10 64 bit - i5 6500 CPU - GTX970 GPU - **Rift DK2** - No Mac hardware here.

**Bug unrelated to this PR as it's present in dev tree up to 5091 - Snapshots present window element as usual for snapshot, but, it's totally blank - uniform gray.** .

This PR DK2
hifi-snap-by-omegaheron-on-2016-06-30_15-33-57

This PR Desktop
hifi-snap-by-omegaheron-on-2016-06-30_15-34-23

Latest Dev (5091) DK2
hifi-snap-by-omegaheron-on-2016-06-30_15-37-29

Latest Dev (5091) Desktop
hifi-snap-by-omegaheron-on-2016-06-30_15-37-09

**Comment by SamGondelman** [comment]
@OmegaHeron whoops I was logged out so I didn't even notice that!  Thanks, I'll fix it in another PR.

**Pull request merged**